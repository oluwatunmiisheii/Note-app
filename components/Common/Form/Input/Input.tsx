import {
  View,
  TextInput,
  StyleSheet,
  TextInputProps,
  ViewStyle,
  StyleProp,
  Text,
} from "react-native";
import colors from "@/utils/constants/colors";
import { InputError } from "../InputError/InputError";

interface InputProps extends TextInputProps {
  extraStyles?: StyleProp<ViewStyle>;
  label: string;
  error?: string;
}

export const Input = (props: InputProps) => {
  const inputStyles = [styles.input, props.multiline && styles.multiline];

  return (
    <View style={{ marginBottom: 20 }}>
      <Text style={styles.label}>{props.label}</Text>
      <View style={styles.container}>
        <TextInput
          style={[inputStyles, props.extraStyles]}
          accessibilityLabel="input"
          placeholderTextColor="#BDBDBD"
          {...props}
        />
      </View>
      {props.error && <InputError error={props.error} />}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#F5F5F5",
    width: "100%",
    paddingVertical: 5,
    paddingHorizontal: 10,
    borderRadius: 10,
    justifyContent: "space-between",
  },
  label: {
    fontSize: 18,
    fontWeight: "500",
    marginBottom: 5,
    color: colors.gray,
  },
  input: {
    height: 50,
    fontSize: 18,
    fontWeight: "500",
    paddingHorizontal: 2,
    textAlignVertical: "top",
  },
  multiline: {
    height: 250,
  },
});
