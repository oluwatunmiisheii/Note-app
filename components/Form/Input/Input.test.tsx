import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react-native";
import { Input } from "./Input";

const onChangeText = jest.fn();
const value = "Value";
const label = "Label";

const defaultProps: React.ComponentProps<typeof Input> = {
  label,
  value,
  onChangeText,
};

const create = (props: Partial<typeof defaultProps> = defaultProps) => {
  const tree = renderer
    .create(<Input {...{ ...defaultProps, ...props }} />)
    .toJSON();
  return { props, tree };
};

const setup = (props: Partial<typeof defaultProps> = defaultProps) => {
  const utils = render(<Input {...{ ...defaultProps, ...props }} />);
  return { props, ...utils };
};

describe(`<${Input.name}>`, () => {
  test("renders correctly", () => {
    const { tree } = create();
    expect(tree).toMatchSnapshot();
  });

  test("renders label", async () => {
    setup();
    expect(screen.getByText(label)).toBeOnTheScreen();
  });

  test("renders value", async () => {
    setup();
    const input = screen.getByLabelText("input");
    fireEvent.changeText(input, "New Value");
    expect(input.props.value).toBe(value);
  });

  test("should call onChangeText when you type", async () => {
    setup();
    const input = screen.getByLabelText("input");
    fireEvent.changeText(input, "New Value");
    expect(onChangeText).toHaveBeenCalled();
  });
});
