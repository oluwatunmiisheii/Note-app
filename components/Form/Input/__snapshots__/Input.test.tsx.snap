// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#6c757d",
        "fontSize": 18,
        "fontWeight": "500",
        "marginBottom": 5,
      }
    }
  >
    Label
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "borderRadius": 10,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      accessibilityLabel="input"
      label="Label"
      onChangeText={[MockFunction]}
      placeholderTextColor="#BDBDBD"
      style={
        Array [
          Array [
            Object {
              "fontSize": 18,
              "fontWeight": "700",
              "height": 50,
              "paddingHorizontal": 2,
              "textAlignVertical": "top",
            },
            undefined,
          ],
          undefined,
        ]
      }
      value="Value"
    />
  </View>
</View>
`;
